lib/utils.js

```diff
@@ -137,38 +157,39 @@ module.exports = class AccessUtils {
       return;
     }
 
-    debug('Attaching before save observer to %s', modelName);
+    winston.debug(`BEFORE SAVE observer - Attaching before save observer to ${modelName}`);
     Model.observe('before save', (ctx, next) => {
+
+      // Only update GroupId  on new instances
       if (!ctx.isNewInstance) {
-        debug('updating %s with params - %o', modelName, ctx.data);
         return next();
       }
 
-      debug('original object before save %o', ctx.instance);
+      winston.debug(`BEFORE SAVE observer - Original data before save ctx.instance ${util.inspect(ctx.instance)}`);
+      winston.debug(`BEFORE SAVE observer - Original data before save ctx.data ${util.inspect(ctx.data)}`);
 
       const currentUser = this.getCurrentUser();
       const foreignKey = this.options.foreignKey;
 
       if (currentUser) {
         // Do not filter if options.skipAccess has been set.
         if (ctx.options.skipAccess) {
-          debug('skipAccess: true - skipping access filters');
+          winston.debug(`BEFORE SAVE observer - skipAccess: true, don't update ${foreignKey} for ${modelName} ${JSON.stringify(ctx.query)}`);
           return next();
         }
 
         // TODO (Alex): the code below assumes the foreignKey exists on the User model.
         // This has been written specifically for our app
         if (ctx.instance[foreignKey] && ctx.instance[foreignKey] !== currentUser[foreignKey]) {
-          debug('object field %s - %s does not match currentUser - %s',
-            foreignKey, ctx.instance[foreignKey], currentUser[foreignKey]);
+          winston.warn(`BEFORE SAVE observer - Object field ${foreignKey} - ${ctx.instance[foreignKey]} does not match currentUser - ${currentUser[foreignKey]}`);
         }
 
-        debug('setting object field %s to %s',
-          foreignKey, currentUser[foreignKey]);
-
+        winston.debug(`BEFORE SAVE observer - Setting object field ${foreignKey} to ${currentUser[foreignKey]}`);
         if (foreignKey && currentUser[foreignKey]) {
           ctx.instance[foreignKey] = currentUser[foreignKey];
```

Comment:
i wonder if we should comment out this line for now and just log what the order's orgId would've been set as? or only set it if `ctx.instance[foreignKey]` is undefined or null?